contracts/token/CRC1155/presets/CRC1155PresetAutoId.sol

```diff
@@ -0,0 +1,158 @@
+        mint(to, 1);
+    }
+
+    /**
+     * @dev Creates `amount` new tokens for `to` with auto-generated token ID.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mint(address to, uint256 amount) public virtual {
+        mint(to, amount, "");
+    }
+
+    /**
+     * @dev Creates `amount` new tokens for `to` with auto-generated token ID and additional data.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mint(address to, uint256 amount, bytes memory data) public virtual {
+        mint(to, _tokenIdTracker.current(), amount, data);
+        _tokenIdTracker.increment();
+    }
+
+    /**
+     * @dev Creates tokens with auto-generated token IDs in batch.
+     * @param to address to mint tokens.
+     * @param count number of token IDs to mint, each token id has amount of 1.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mintBatch(address to, uint256 count) public virtual {
+        uint256[] memory amounts = new uint256[](count);
+
+        for (uint i = 0; i < count; i++) {
+            amounts[i] = 1;
+        }
+
+        mintBatch(to, amounts);
+    }
+
+    /**
+     * @dev Creates tokens with auto-generated token IDs in batch.
+     * @param to address to mint tokens.
+     * @param amounts amount for each token ID to mint.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mintBatch(address to, uint256[] memory amounts) public virtual {
+        mintBatch(to, amounts, "");
+    }
+
+    /**
+     * @dev Creates tokens with auto-generated token IDs in batch.
+     * @param to address to mint tokens.
+     * @param amounts amount for each token ID to mint.
+     * @param data Additional data used when `to` address is a contract.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mintBatch(address to, uint256[] memory amounts, bytes memory data) public virtual {
+        uint256[] memory tokenIds = new uint256[](amounts.length);
+
+        for (uint i = 0; i < amounts.length; i++) {
+            tokenIds[i] = _tokenIdTracker.current();
+            _tokenIdTracker.increment();
+        }
+
+        mintBatch(to, tokenIds, amounts, data);
+    }
+
+    /**
+     * @dev See {ERC1155-_beforeTokenTransfer}.
+     *
+     * Requirements:
+     *
+     * - each value in `amounts` should not be zero.
+     */
+    function _beforeTokenTransfer(
+        address operator,
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    )
+        internal virtual override(ERC1155PresetMinterPauser, CRC1155Enumerable)
+    {
+        for (uint i = 0; i < amounts.length; i++) {
+            require(amounts[i] > 0, "CRC1155Preset: amount is zero");
+        }
```

Comment:
Why is it useful to add this check here?

Reply:
To prevent transfer/mint/burn 0 token by mistake, it comes from the community.